### PR TITLE
[SourceKit] Delete old C++ `NameMatcher`

### PR DESCRIPTION
**Pull request by ahoppen**
Replace all usages of the old C++ `NameMatcher` with the `NameMatcher` in swift-syntax.

rdar://118994505

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

**Comment by ahoppen** [comment]
@swift-ci Please SourceKit stress test

**Review by bnbarham — APPROVED** [review]

File: include/swift/IDE/IDEBridging.h [hunk 1]
```diff
@@ -163,8 +163,10 @@ class BridgedResolvedLocVector {
 extern "C" {
 #endif
 
-/// Entry point to run the NameMatcher written in swift-syntax.
-/// 
+/// Low-level entry point to run the NameMatcher written in swift-syntax.
+///
+/// - Warning: The only caller of this should be `swift::ide::runNameMatcher`.
```
Comment (hamishknight) [inline include/swift/IDE/IDEBridging.h:168]: Would it be worth moving this into the cpp file in that case?

**Review by hamishknight — APPROVED** [review]

**Comment by ahoppen** [comment]
Note to self: The stress tester passed

File: lib/IDE/CMakeLists.txt [hunk 2]
```diff
@@ -46,4 +46,14 @@ target_link_libraries(swiftIDE PRIVATE
   swiftParse
   swiftSema)
 
+if (SWIFT_BUILD_SWIFT_SYNTAX)
+  target_compile_definitions(swiftIDE
+    PRIVATE
+    SWIFT_BUILD_SWIFT_SYNTAX
+  )
```
[the remark this answers is not in the document]
Comment (ahoppen) [reply lib/IDE/CMakeLists.txt:53]: Good idea. Moved it.

File: include/swift/IDE/IDEBridging.h (hunk 1 above)
Comment (ahoppen) [reply include/swift/IDE/IDEBridging.h:168]: Good idea 👍🏽

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

**Comment by ahoppen** [comment]
@swift-ci Please test Linux

**Comment by ahoppen** [comment]
@swift-ci Please test Linux

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Pull request merged**